### PR TITLE
Add merge methods

### PR DESCRIPTION
**Pull request by 0mbi**

### CODE REVIEW

**Comment by Mic92** [comment]
Can you rebase?

File: nixpkgs_review/cli/__init__.py [hunk 1]
```diff
@@ -194,6 +194,7 @@ def parse_args(command: str, args: List[str]) -> argparse.Namespace:
     comments_parser.set_defaults(func=show_comments)
 
     merge_parser = subparsers.add_parser("merge", help="merge PR")
+    merge_parser.add_argument("-m", "--method", default="merge", help="Merge method")
```
Comment (Mic92) [inline nixpkgs_review/cli/__init__.py:197]: Did you forget to commit a file? `merge_parser.method` is not used anywhere.

Comment (0mbi) [reply nixpkgs_review/cli/__init__.py:197]: Fixing it soon. Setting WIP for now!

Comment (0mbi) [reply nixpkgs_review/cli/__init__.py:197]: Ok sorry, for the long wait. Should be fixed now.

File: nixpkgs_review/cli/merge.py [hunk 2]
```diff
@@ -6,4 +6,4 @@
 
 def merge_command(args: argparse.Namespace) -> None:
     github_client = GithubClient(ensure_github_token(args.token))
-    github_client.merge_pr(get_current_pr())
+    github_client.merge_pr(get_current_pr(), dict(method=args.method))
```
Comment (Mic92) [inline nixpkgs_review/cli/merge.py:9]: Have you tested if that works? It's called `merge_method` here: https://docs.github.com/en/rest/reference/pulls#merge-a-pull-request

File: nixpkgs_review/github.py [hunk 3]
```diff
@@ -49,13 +49,13 @@ def approve_pr(self, pr: int) -> Any:
         "Approve a PR"
         print(f"Approving {pr_url(pr)}")
         return self.post(
-            f"/repos/NixOS/nixpkgs/pulls/{pr}/reviews", data=dict(event="APPROVE"),
+            f"/repos/NixOS/nixpkgs/pulls/{pr}/reviews", data=dict(event="APPROVE")
         )
 
-    def merge_pr(self, pr: int) -> Any:
+    def merge_pr(self, pr: int, data: Optional[Dict[str, str]] = None) -> Any:
```
Comment (Mic92) [inline nixpkgs_review/github.py:55]: Expose this as an optional argument instead for more type safety:

```suggestion
    def merge_pr(self, pr: int, method_method: str = "merge") -> Any:
```

File: nixpkgs_review/github.py [hunk 4]
```diff
@@ -49,13 +49,13 @@ def approve_pr(self, pr: int) -> Any:
         "Approve a PR"
         print(f"Approving {pr_url(pr)}")
         return self.post(
-            f"/repos/NixOS/nixpkgs/pulls/{pr}/reviews", data=dict(event="APPROVE"),
+            f"/repos/NixOS/nixpkgs/pulls/{pr}/reviews", data=dict(event="APPROVE")
         )
 
-    def merge_pr(self, pr: int) -> Any:
+        def merge_pr(self, pr: int, merge_method: str = "merge") -> Any:
```
Comment (Mic92) [inline nixpkgs_review/github.py:55]:
```suggestion
      def merge_pr(self, pr: int, merge_method: str = "merge") -> Any:
```

File: nixpkgs_review/cli/__init__.py (hunk 1 above)
Comment (Mic92) [inline nixpkgs_review/cli/__init__.py:197]:
```suggestion
    merge_parser.add_argument("-m", "--method", default="merge", help="Merge method", choices=["merge", "squash", "rebase"])
```

File: nixpkgs_review/cli/merge.py (hunk 2 above)
Comment (Mic92) [reply nixpkgs_review/cli/merge.py]:
```
curl \
  -X PUT \
  -H "Accept: application/vnd.github.v3+json" \
  https://api.github.com/repos/octocat/hello-world/pulls/42/merge \
  -d '{"commit_title":"commit_title"}'
```

**Comment by Mic92** [comment]
Closing it for now. Please re-open if you have time again to work on it.

**Pull request closed without merging**